machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -107,6 +114,41 @@ type MachineConfigNodeStatus struct {
 	// This desired machine config has been compared to the current machine config and has been validated by the machine config operator as one that is valid and that exists.
 	// +kubebuilder:validation:Required
 	ConfigVersion MachineConfigNodeStatusMachineConfigVersion `json:"configVersion"`
+	// pinnedImageSets describes the current and desired pinned image sets for this node.
+	// The current version is the generation of the pinned image set that has most recently been successfully pulled and pinned on this node.
+	// The desired version is the generation of the pinned image set that is targeted to be pulled and pinned on this node.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	PinnedImageSets []MachineConfigNodeStatusPinnedImageSet `json:"pinnedImageSets,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.desiredGeneration) ? self.desiredGeneration > self.currentGeneration : true",message="desired generation must be greater than current generation"
```

Comment:
can also be equal

Reply:
> can also be equal when you're updating the currentGenreation, right?

Probably worth adding the tests to the integration suite so we can be sure.

Reply:
added  status update tests to cover